### ISSUE TITLE
Fails to run in Safari 15.4 on iOS 

### ISSUE THREAD

**Issue by Herschel** [ruffle-rs/ruffle#6608]
Currently Ruffle crashes immediately in Safari 15.4 on iOS due to a WebAssembly miscompilation bug in Safari.

This typically results in a memory-related error, such as "Out of bounds memory access" or similar errors during the tessellation steps.

This is awaiting a fix on Apple's side:
https://bugs.webkit.org/show_bug.cgi?id=237180
https://bugs.webkit.org/show_bug.cgi?id=238302

**Comment by pestoforce** [comment]
Reporting the panic error happening on my iPad here as well:
```
# Error Info
Error name: Error
Error message: panicked at 'Bezier without endpoint', core/src/shape_utils.rs:213:36
Error stack:

@https://typehelp.org/ruffletest/ruffle/core.ruffle.35ba130500fa865e8e67.js:1:7717
wasm-stub@[wasm code]
<?>.wasm-function[std::sync::once::Once::call_once::{{closure}}::h802bfeb6f8bb5638]@[wasm code]
<?>.wasm-function[std::sync::once::Once::call_inner::h07d0245bef259c27]@[wasm code]
<?>.wasm-function[core::ops::function::FnOnce::call_once{{vtable.shim}}::h9ecdeaaea35085f8]@[wasm code]
<?>.wasm-function[std::panicking::rust_panic_with_hook::h623ac32ff431b114]@[wasm code]
<?>.wasm-function[core::panicking::panic_fmt::h5118e89563022e7e]@[wasm code]
<?>.wasm-function[core::option::expect_failed::h52436cb58dfe4ec3]@[wasm code]
<?>.wasm-function[<core::iter::adapters::flatten::FlatMap<I,U,F> as core::iter::traits::iterator::Iterator>::next::hed214b29a827cdd6]@[wasm code]
<?>.wasm-function[ruffle_core::shape_utils::ShapeConverter::flush_layer::h66beb67534a7dd6e]@[wasm code]
<?>.wasm-function[<ruffle_core::shape_utils::DistilledShape as core::convert::From<&swf::types::Shape>>::from::h6e92fa6608141852]@[wasm code]
<?>.wasm-function[ruffle_core::display_object::movie_clip::MovieClipData::define_shape::ha98dc4b982c17625]@[wasm code]
<?>.wasm-function[ruffle_core::display_object::movie_clip::MovieClip::preload::hbaf6f019c2eaff27]@[wasm code]
<?>.wasm-function[ruffle_core::player::Player::set_root_movie::h1a20254dba634c51]@[wasm code]
<?>.wasm-function[<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h903ee168fc90b9ee]@[wasm code]
<?>.wasm-function[<core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h33ed72626bb34236]@[wasm code]
<?>.wasm-function[wasm_bindgen_futures::queue::Queue::new::{{closure}}::he8e3ce0ab6352214]@[wasm code]
<?>.wasm-function[<dyn core::ops::function::FnMut<(A,)>+Output = R as wasm_bindgen::closure::WasmClosure>::describe::invoke::h40f6ecbecfe0d2cc]@[wasm code]
wasm-stub@[native code]
O@https://typehelp.org/ruffletest/ruffle/core.ruffle.35ba130500fa865e8e67.js:1:3868
c@https://typehelp.org/ruffletest/ruffle/core.ruffle.35ba130500fa865e8e67.js:1:2393
promiseReactionJob@[native code]

# Player Info
Allows script access: true
Player type: Object
SWF URL: pdtd.swf
Param movie: pdtd.swf
Param allowscriptaccess: always
Attribute 0: undefined
Attribute 1: undefined
Attribute 2: undefined
Attribute 3: undefined

# Page Info
Page URL: https://typehelp.org/ruffletest/pdtd.html
SWF URL: https://typehelp.org/ruffletest/pdtd.swf

# Browser Info
Useragent: Mozilla/5.0 (iPad; CPU OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/100.0.4896.85 Mobile/15E148 Safari/604.1
OS: iPad

# Ruffle Info
Version: 0.1.0
Name: nightly 2022-04-14
Channel: nightly
Built: 2022-04-14T00:15:14.709Z
Commit: 036aa2b369aa9105800fbd49954bd2ae42281a0a
Is extension: false
```

**Comment by rdrpenguin04** [comment]
Both of the linked bug reports are marked as fixed; do we wait for the next Safari update?

**Comment by Herschel** [comment]
It's fixed in their development repo but we still have to wait for the ship to fix to the public.

I just tried iPad OS 15.5 beta 3 and unfortunately still have the same issues, so not sure when exactly that will be.

**Comment by Herschel** [comment]
The fix seems to have made it into iOS 15.5 beta 4, released today (May 3). 😄 

Now we just have to wait for the final release of 15.5, which should be sometime this month.

**Comment by Herschel** [comment]
iOS 15.5 released today (May 16) which fixes these issues.
